superset/assets/src/visualizations/core/components/SuperChart.jsx

```diff
@@ -0,0 +1,167 @@
+  overrideTransformProps: undefined,
+  postTransformProps: IDENTITY,
+  onRenderSuccess() {},
+  onRenderFailure() {},
+  skipRendering: false,
+};
+
+class SuperChart extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.renderChart = this.renderChart.bind(this);
+    this.renderLoading = this.renderLoading.bind(this);
+
+    // memoized function so it will not recompute
+    // and return previous value
+    // unless one of
+    // - preTransformProps
+    // - transformProps
+    // - postTransformProps
+    // - chartProps
+    // is changed.
+    this.processChartProps = createSelector(
+      input => input.preTransformProps,
+      input => input.transformProps,
+      input => input.postTransformProps,
+      input => input.chartProps,
+      (pre, transform, post, chartProps) => post(transform(pre(chartProps))),
+    );
+
+    const componentRegistry = getChartComponentRegistry();
+    const transformPropsRegistry = getChartTransformPropsRegistry();
+
+    // memoized function so it will not recompute
+    // and return previous value
+    // unless one of
+    // - chartType
+    // - overrideTransformProps
+    // is changed.
+    this.createLoadableRenderer = createSelector(
+      input => input.chartType,
+      input => input.overrideTransformProps,
+      (chartType, overrideTransformProps) => {
+        if (chartType) {
+          return Loadable.Map({
+            loader: {
+              Chart: () => componentRegistry.getAsPromise(chartType),
+              transformProps: overrideTransformProps
+                ? () => Promise.resolve(overrideTransformProps)
+                : () => transformPropsRegistry.getAsPromise(chartType),
+            },
+            loading: loadingProps => this.renderLoading(loadingProps, chartType),
+            render: this.renderChart,
+          });
+        }
+        return null;
+      },
+    );
+  }
+
+  renderChart(loaded, props) {
+    const Chart = loaded.Chart.default || loaded.Chart;
+    const transformProps = loaded.transformProps;
+    const {
+      chartProps,
+      preTransformProps,
+      postTransformProps,
+    } = props;
+
+    const result = (
+      <Chart
+        {...this.processChartProps({
+          preTransformProps,
+          transformProps,
+          postTransformProps,
+          chartProps,
+        })}
+      />
+    );
+    setTimeout(() => this.props.onRenderSuccess(), 0);
+    return result;
+  }
+
+  renderLoading(loadableProps, chartType) {
+    const { error, pastDelay } = loadableProps;
+
+    if (error) {
+      const result = (
+        <div className="alert alert-warning" role="alert">
+          <strong>ERROR</strong>&nbsp;
+          <code>chartType="{chartType}"</code> &mdash;
+          {JSON.stringify(error)}
+        </div>
+      );
+      setTimeout(() => this.props.onRenderFailure(error), 0);
+      return result;
+    } else if (pastDelay) {
+      return (
+        <span>Loading...</span>
```

Comment:
What do you think about removing this? I would guess that the case where the chart **data** has come back in `< 200ms` (the `Loadable` `delay` default) and the chart **code** has not is rare. 

This would also print `Loading...` on top of the existing Superset `<Loader />` if chart data is still loading and chart code takes `> 200ms` to load yeah?